core/domain/html_cleaner.py

```diff
@@ -182,7 +182,15 @@ def unescape_html(escaped_html_data):
     Returns:
         str. Unescaped HTML string.
     """
-    unescaped_html_data = escaped_html_data
+    # Some of the explorations having font-family in style tags
+    # include double quotes around the font name. This results
+    # in error while performing json.loads.
+    # Any such extra quotes are not rendered as \&quot; in html
+    # string. They are rendered in form of &amp;quot;. This line
+    # ensures that all such extra quotes are replaced with
+    # \&amp;quot; in the html string.
+    unescaped_html_data = escaped_html_data.replace(
+        '&amp;quot;', '\\&amp;quot;')
```

Comment:
> The only way is that we replace &amp;quot; only if it is after font-family attribute in style tags.

If you are sure that this happens only in font-family then include font-family in the check. The more specific the better.

Also I have a question about the current code. What if the data already contains `\\&amp;quot;`?

Reply:
If data contains `\\&amp;quot;` it is rendered as `\\&quot;` in the string before unescaping. The quotes after font-family in data are as `&amp;amp;quot;`, so they are rendered as `&amp;quot;` in string before unescaping. So, data will not contain `\\&amp;quot;`.

Reply:
The other thing we can do is that `\\` only if they are not present.

Reply:
That would be better.

Reply:
What do you suggest here @seanlip @AllanYangZhou ?

Reply:
I thought I just made a suggestion? The stricter the check, the better.

Reply:
Alternatively you can do, before this line, something like: "if string_we're_not_supposed_to_find in data: then error"

Reply:
We have two options to make it strict:
- Check for font-family before `&amp;quot;`
- Replace `&amp;quot;` with `\&amp;quot;` only if `\` is missing

I feel it is better to go with second since it will fix this mistake if it is anywhere else as well?

Reply:
Do both?

Reply:
(Though if you have to pick one I lean towards the first. Reasoning: if it was anywhere else, it would have shown up already, right?

At this point I think it's more important to prevent breaking other stuff.)

Reply:
For the first check, I need to find the font-family attribute and then look for opening closing `&amp;quot;`. Is there no way except regex for this? 

It's a bit complicated to write regex for the same since there can be multiple pairs of `&amp;quot;`.
```
font-family:&amp;quot;Times New Roman&amp;quot;;\nmso-fareast-font-family:&amp;quot;Times New Roman&amp;quot;
```
Here there are two pairs of `&amp;quot;` and I want to fetch them.

Reply:
I don't think I understand the problem. Why not search for `font-family:&amp;quot;` and replace that string?

Reply:
There is an `&amp;quot` in the end as well. 
`font-family:&amp;quot;Times New Roman&amp;quot;`

Reply:
Oh, I see, it's both the start and end?

Should be able to do something like `font-family:&amp;quot;[...]+&amp;quot;`, right? See https://docs.python.org/2/library/re.html#re.sub too.

Reply:
(you'll need to figure out what goes in "...")